app/src/main/java/io/github/sheasmith/discountdetective/ui/screens/ShoppingListScreen.kt

```diff
@@ -176,21 +187,21 @@ private fun ProductList(
 
 @Composable
 private fun ProductCard(
-    product: Triple<RetailerProductInformation, StorePricingInformation, Int>,
-    elevation: State<Dp>
+    product: Triple<RetailerProductInformation, StorePricingInformation, ShoppingListItem>,
+    elevation: State<Dp>,
+    checked: Boolean,
+    onCheckedChanged: (Boolean) -> Unit,
+    onDelete: () -> Unit
 ) {
-    //checkbox Checked
-    val isChecked = remember { mutableStateOf(false) }
+    // remember local action state
+    var isChecked by rememberSaveable { mutableStateOf(checked) }
```

Comment:
Not sure exactly why you need this? This function will be called whenever `checked` is updated anyway, so this will never be used.

Reply:
The mutablestateFlow only listens for insertions or deletions. You can;t add a mutableState to the dataclass for checked because room doesn't like it/with that join we do with all the repos it'd be ugly. So need this here to update the page while a user is on the page ( otherwise only will update on page change).